### ISSUE TITLE
back gesture from Flutter Gallery > Text Fields does not work on iOS?

### ISSUE THREAD

**Issue by eseidelGoogle** [flutter/flutter#9200]
No clue why.

FYI @xster

**Comment by xster** [comment]
I think it wasn't a regression. The code checked for the presence of a onWillPop overrider (which the demo has) and disables the back gesture altogether rather than checking the return value of the onWillPop handler at the time of the gesture. 

Maybe the it was designed so originally to avoid awkward UI irresponsiveness and unexpected behaviour since the user can inject an onWillPop handler and do whatever he/she wants inside. For instance, the demo shows a dialog saying 'this form has errors, do you really want to leave'. I think a similar 'infallible instant pop' expectation from the user probably applies to the back button itself as well though. 

Maybe we should make the recommendation to the user that if the user wants to override the poppability of a page, he/she should present the page as a fullscreen dialog (with an 'X' and a 'Submit'/'Done'/'Save' button) instead of a normal stack page.

**Comment by Hixie** [comment]
It should be an error to have an onWillPop overrider if the route supports something like the back gesture, IMHO.

**Comment by xster** [comment]
Agreed. The existing implementation kinda does what you're saying. Back gesture is enabled by default (which I think is fine for not asking the user to explicitly specify too many things) and turns off when there's an overrider. 

I think the only thing 'wrong' was the demo uses a dialog behaviour (confirmation before leaving) without presenting a dialog affordance. I'll see if I can tweak the scaffold/appbar to automate some of this stuff.

**Comment by xster** [comment]
Ok, to make easier for users to not forget to go to the fullscreen dialog's app bar and swap out the leading button to an X instead of the default <, I'll move the fullscreenDialog knowledge from MaterialPageRoute to ModalRoute, next to barrierDismissible etc which I think scopes nicely. 

The AppBar will read it and present an appropriate button. Has the added benefit of letting it centrally drive other fullscreen dialog related UI like a white appbar background for iOS.

**Comment by xster** [comment]
Remaining: let the text demo page have a view mode and an edit mode

**Comment by xster** [comment]
I'm blocked on this. The text demo is actually broken on iOS until we fix #9511 or #9513.

**Comment by eseidelGoogle** [comment]
Doesn't seem like this is required for M2?

**Comment by guidezpl** [comment]
Closing as obsolete
